### PR TITLE
Fix/update resource name display and fix misc bugs

### PR DESCRIPTION
**Pull request by alexanderleegs**
This PR resolves https://github.com/isomerpages/isomercms-frontend/issues/292. It updates our resource name display in preparation for the resource name migration to rename existing resource files. The title of resources are now prettified, similarly to what is being done for pages. Additionally, we check for the existence of the special string "dollar" followed by any alphanumeric character, as this indicates that the title of the file contains a $ sign. The reason we require this is due to the generation and prettifying of resource file names - the filename will still contain the dollar string, but we would like the display the title on the cms as the user first inputted it.

In addition, the following bugs have been fixed:
- A typo in the endpoint being called when deleting folders has been corrected.
- Previously, we were not distinguishing correctly between new and existing collections, causing an error when we attempted to retrieve collection data for a new collection. A new variable to check for new collections has been added to OverviewCard.
- The title of the delete warning modal when deleting pages has been corrected.

### CODE REVIEW

File: src/components/OverviewCard.jsx [hunk 1]
```diff
@@ -74,7 +75,7 @@ const OverviewCard = ({
       } = frontMatter;
 
       let collectionPageData
-      if (!isResource && chosenCategory) {
+      if (!isResource && !isNewCollection) {
         // User selected an existing page collection
         const { collectionPages } = await retrieveThirdNavOptions(siteName, chosenCategory, true)
         collectionPageData = collectionPages
```
Comment (kwajiehao) [inline src/components/OverviewCard.jsx:81]: Currently, this fails to account for the case when we attempt to move a page from a collection back to unlinked page

**Review by kwajiehao — CHANGES_REQUESTED** [review]
left comments

File: src/utils.js [hunk 2]
```diff
@@ -107,7 +107,16 @@ export function retrieveResourceFileMetadata(fileName) {
...
+  const titleTokenArray = tokenArray.slice(3);
+  const prettifiedTitleTokenArray = titleTokenArray.map((token) => {
+    if (token.length < 2) return token.toUpperCase()
+    return token.slice(0,1).toUpperCase() + token.slice(1)
+  });
+  const prettifiedTitle = prettifiedTitleTokenArray.join(' ')
+
+  // We search for special characters which were converted to text
+  // Convert dollar back to $ if it is followed by any alphanumeric character
+  const title = prettifiedTitle.replaceAll(/dollar(?=([a-zA-Z0-9]))/gi, '$')
```
[the remark this answers is not in the document]
Comment (alexanderleegs) [reply src/utils.js]: Yeah that makes more sense actually, fixed in https://github.com/isomerpages/isomercms-frontend/pull/294/commits/335ab12b00f3afe5b46206ac92bb62fbfb0fdf60

File: src/components/OverviewCard.jsx (hunk 1 above)
Comment (alexanderleegs) [reply src/components/OverviewCard.jsx]: Good catch, resolved in https://github.com/isomerpages/isomercms-frontend/pull/294/commits/79bb5f447355ece52ed299aed6e8578a2bf94378

**Review by kwajiehao — APPROVED** [review]
lgtm

**Pull request merged**